src/main/java/uk/gov/nationalarchives/pdi/step/jena/model/JenaModelStepMeta.java

```diff
@@ -107,6 +108,7 @@
         String fieldName;
         QName rdfPropertyName;
         QName rdfType;
+        String language; // TODO(SL): Should this be java.util.Locale?
```

Comment:
Is the language optional? If not what is the default?

If it is optional we should probably mark it as `@Nullable` and with the serialization/deserialization we need to handle that.

Reply:
> Is the language optional? If not what is the default?

Language is optional.

> If it is optional we should probably mark it as `@Nullable`

I tried annotating with `@Nullable`, Kettle doesn't seem to care either way.
Should I mark it?
How about others fields on this class?
How should I annotate required fields with (if at all)?
In my understanding, field name and property name are required, type and language are optional.

> with the serialization/deserialization we need to handle that.

Serialisation/deserialisation handles both cases:
https://github.com/nationalarchives/kettle-jena-plugins/blob/d95bb22a2ec7f55f4c4934df632fe9d694fc65e2/src/main/java/uk/gov/nationalarchives/pdi/step/jena/model/JenaModelStep.java#L240-L246

Reply:
It's in general good form to annotate potentially null fields, parameters, and return values with `javax.annotation.Nullable` i.e. `@Nullable`.

Two quick good reasons:

1. It serves as a form of code documentation. The caller of a function or accessor to a field knows what to expect.
2. It can be used in conjunction with Static code checkers like FindBugs or PMD to improve code quality through automated review.